app/frontend/src/features/profile/edit/ProfileForm.tsx

```diff
@@ -76,6 +105,9 @@ export default function EditProfileForm() {
   const onSubmit = handleSubmit((data) => {
     resetUpdate();
     updateUserProfile({ profileData: data, setMutationError: setErrorMessage });
+    if (!errorMessage) {
```

Comment:
`updateUserProfile` is an async operation, but the mutation exposed is not so this will always succeed as `errorMessage` wouldn't be populated at this time. The mutation exposed is not async here as all of the logic (`onSuccess`, `onError` etc) is inside the hook that exposes this. Might be worth moving this into `onSuccess` in `useUpdateUserProfile` so the logic remain in one place.